src/ring_theory/integral_closure.lean

```diff
@@ -297,6 +297,37 @@ lemma algebra.is_integral_trans (A_int : ∀ x : A, is_integral R x)(B_int : ∀
   ∀ x:B, is_integral R x :=
 λ x, is_integral_trans A_int x (B_int x)
 
+lemma is_integral_of_surjective (h : function.surjective (algebra_map R A))
+  : ∀ x : A, is_integral R x :=
+λ x, (h x).rec_on (λ y hy, (hy ▸ is_integral_algebra_map : is_integral R x))
+
+/-- If `R → A → B` is an algebra tower with `A → B` injective,
+then if the entire tower is an integral extension so is `R → A` -/
+lemma is_integral_of_is_integral_tower_back (H : function.injective (algebra_map A B))
```

Comment:
Thanks for the PR. Looks good to me, except that I don't like the `_back` and `_front` names. (My initial guess would have put them on the wrong statement.) Can we think of better suffixes? How about the `_bot` and `_top` of a tower? Or is that to confusing with lattice nomenclature? @kckennylau @Vierkantor do you have opinions?
Other options that come to mind:

* `_pre` and `_post`
* `_fst` and `_snd` or `_fst` and `_last`
* `_left` and `_right`

The benefit of `_bot` and `_top` is that `_tower` already suggests some vertical intuition (which is also very common in informal presentations on blackboards, of course), and so I think intuitively, everyone agrees on what the `_bot` and `_top` of a tower are...

Reply:
I vote for `_bot` and `_top`. They would even be the order bottom/top according to the inclusion order on `R → A → B`.

Reply:
I went ahead and switched to `_bot` and `_top` for now while fixing the colon issue, I also think it makes the most sense with the vertical naming of tower.